test/bin/jar.spec.ts

```diff
@@ -0,0 +1,26 @@
+import jar from "keycloakify/bin/tools/jar";
+import { it, describe, vi } from "vitest";
+
+vi.mock("fs", () => ({ promises: { mkdir: () => {} }, createWriteStream: () => {} }));
+vi.mock("stream", async () => {
+    const readableMock = () => {
+        const mockDecorators = {
+            on: () => mockDecorators,
+            pipe: () => mockDecorators
+        };
+        return {
+            from: () => mockDecorators
+        };
+    };
+
+    return {
+        // @ts-ignore
+        ...(await vi.importActual("stream")),
+        Readable: readableMock()
+    };
+});
+describe("jar", () => {
```

Comment:
I thought this test was going to look cleaner, but after mocking imports I'm not sure how valuable it is. I can remove if wanted.